packages/storage/src/indexed-db/index.ts

```diff
@@ -458,12 +458,14 @@ export class IndexedDb implements IndexedDbInterface {
     });
   }
 
-  async addEpoch(startHeight: bigint, index?: bigint): Promise<void> {
-    if (index === undefined) {
-      const cursor = await this.db.transaction('EPOCHS', 'readonly').store.openCursor(null, 'prev');
-      const previousEpoch = cursor?.value ? Epoch.fromJson(cursor.value) : undefined;
-      index = previousEpoch?.index !== undefined ? previousEpoch.index + 1n : 0n;
-    }
+  /**
+   * Adds a new epoch with the given start height. Automatically sets the epoch
+   * index by finding the previous epoch index, and adding 1n.
+   */
+  async addEpoch(startHeight: bigint): Promise<void> {
+    const cursor = await this.db.transaction('EPOCHS', 'readonly').store.openCursor(null, 'prev');
+    const previousEpoch = cursor?.value ? Epoch.fromJson(cursor.value) : undefined;
+    const index = previousEpoch?.index !== undefined ? previousEpoch.index + 1n : 0n;
```

Comment:
Can we use an async iterator here?

```typescript
  async addEpoch(startHeight: bigint): Promise<void> {
    for await (const cursor of this.db.transaction('EPOCHS').store) {
      const previousEpoch = Epoch.fromJson(cursor.value);
      const index = previousEpoch.index + 1n;
      const newEpoch = {
        startHeight: startHeight.toString(),
        index: index.toString(),
      };
      await this.u.update({
        table: 'EPOCHS',
        value: newEpoch,
      });
    }
  }
```

Reply:
think getEpochByHeight() can be converted to one too

Reply:
Hm, why do it that way? I only want to grab the very last epoch from the table, not iterate over all of them. I guess I could just add a `break` or `return` statement at the end of the first run through the loop, but that feels a bit unidiomatic — people reading this code in the future will initially see it as a loop, and then realize the code is only running once.

Reply:
(I did update `getEpochByHeight` to use this syntax though, per your suggestion.)

Reply:
Ah, didn't realize that was happening. Can you add a code comment? :)

Reply:
Oh wait there is one already lol whoops